ports/curl/portfile.cmake

```diff
@@ -87,6 +89,15 @@ else ()
     list(APPEND BUILD_OPTIONS -DCURL_DISABLE_COOKIES=ON)
 endif ()
 
+# Check for http3 feature
+if (http3 IN_LIST FEATURES)
+    message(STATUS "Enabling HTTP/3 support")
+    list(APPEND BUILD_OPTIONS -DUSE_NGTCP2=ON)
+else ()
+    message(STATUS "Disabling HTTP/3 support")
+    list(APPEND BUILD_OPTIONS -DUSE_NGTCP2=OFF)
+endif ()

```

Comment:
Note that the error checking ends up coming from `ngtcp2` since it depends on `boringssl`. So if someone tried to do `curl[libressl,http3]` it would end up hitting the error state at the start of the `libressl` portfile https://github.com/WebKitForWindows/WebKitRequirements/blob/main/ports/libressl/portfile.cmake#L1-L3